server/audit_math/raire_utils.py

```diff
@@ -328,7 +324,9 @@ def is_descendent_of(self, node):
         if len1 <= len2:
             return False
 
-        return self.tail[len1 - len2 :] == node.tail
+        # More typechecker shenanigans
+        ret_val = bool(self.tail[len1 - len2 :] == node.tail)
+        return ret_val
```

Comment:
This is surprising! I did a little research and it turns out any operator applied to `Any` results in `Any`, and `node` here has type `Any`. I think a better fix for this would be to add a type annotation: `node: RaireNode`.